gothic/gothic.go

```diff
@@ -99,6 +99,10 @@ var SetState = func(req *http.Request) string {
 // This is used to prevent CSRF attacks, see
 // http://tools.ietf.org/html/rfc6749#section-10.12
 var GetState = func(req *http.Request) string {
+	params := req.URL.Query()
+	if params.Encode() == "" && req.Method == http.MethodPost {
+		return req.FormValue("state")
+	}
 	return req.URL.Query().Get("state")
```

Comment:
Since `params` is now define above, you could also do the following for the return.
```suggestion
	return params.Get("state")
```

Reply:
Fixed, thanks